### PR TITLE
Move operators to separate, static dali_kernels.a lib

### PR DESCRIPTION
**Pull request by JanuszL**
Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

### CODE REVIEW

**Comment by JanuszL** [comment]
Builds 595564

**Comment by JanuszL** [comment]
Rebuild 596166.

**Comment by JanuszL** [comment]
Rebuilds 596188

File: dali/CMakeLists.txt [hunk 1]
```diff
@@ -47,16 +49,25 @@ if(${CMAKE_VERSION} VERSION_LESS "3.9.0")
...
+cuda_add_library(${dali_kernel_lib} STATIC ${DALI_KERNEL_SRCS})
+
 cuda_add_library(${dali_lib} SHARED ${DALI_SRCS} ${DALI_PROTO_OBJ})
 
 # Options for using Dockerfile FFmpeg version
 target_compile_definitions(${dali_lib} PRIVATE HAVE_AVSTREAM_CODECPAR=1)
 target_compile_definitions(${dali_lib} PRIVATE HAVE_AVBSFCONTEXT=1)
 
+target_compile_definitions(${dali_kernel_lib} PRIVATE HAVE_AVSTREAM_CODECPAR=1)
+target_compile_definitions(${dali_kernel_lib} PRIVATE HAVE_AVBSFCONTEXT=1)
```
Comment (mzient) [inline dali/CMakeLists.txt]: Why not "dali_operator_lib"?
My point is: operators and kernels are very different beings; they operate on different data types and follow quite different logic. Perhaps when we finally have the kernels as a separate entity, we'd add `dali_kernel_lib`.

Comment (JanuszL) [reply dali/CMakeLists.txt]: Done

**Comment by JanuszL** [comment]
Rebuilds 597098

File: dali/CMakeLists.txt [hunk 2]
```diff
@@ -47,16 +49,25 @@ if(${CMAKE_VERSION} VERSION_LESS "3.9.0")
   # So this is a workaround to append PRIVATE in front of it 
   set(CUDA_LIBRARIES ${CUDA_LINK_LIBRARIES_KEYWORD} ${CUDA_LIBRARIES})
 endif()
+cuda_add_library(${dali_operator_lib} STATIC ${DALI_KERNEL_SRCS})
```
Comment (mzient) [inline dali/CMakeLists.txt]: `DALI_KERNEL_SRCS` -> `DALI_OPERATOR_SRCS`

Comment (JanuszL) [reply dali/CMakeLists.txt]: Done

**Comment by JanuszL** [comment]
Another build 597194

**Pull request merged**